### PR TITLE
Feature/12-ruleset use reducer

### PR DESCRIPTION
**Pull request by jasrial**

### CODE REVIEW

File: src/components/RuleSet.js [hunk 1]
```diff
@@ -8,50 +8,74 @@ import { faTrash } from "@fortawesome/free-solid-svg-icons";
...
-        </Col>
-      </Row>
-    </Container>
-  </div>
-);
+const RuleSet = ({ rules, dimension, onChange, dimensions }) => {
+  const initialState = { dimension, rules };
+  const [state, dispatch] = useReducer(ruleSetReducer, initialState);
+  const dimensionConst = state.dimension;
+  const rulesConst = state.rules;
```
Comment (vlabs) [inline src/components/RuleSet.js]: those two lines can be merged into one:
`const { dimensionConst, rulesConst } = state`

but please rename to `dimensionState` and `rulesState`

File: src/components/RuleSet.js [hunk 2]
```diff
@@ -8,50 +8,74 @@ import { faTrash } from "@fortawesome/free-solid-svg-icons";
...
+            </Row>
+          </div>
+        ))}
+        <Row className="SetRow">
+          <Col md={{ span: 5, offset: 3 }}>
+            <Button
+              variant="outline-secondary"
+              size="lg"
+              block
+              onClick={e => dispatch({ type: "addRule" })}
```
Comment (vlabs) [inline src/components/RuleSet.js]: `e` is unused
```suggestion
              onClick={() => dispatch({ type: "addRule" })}
```

File: src/utils/removeArrayElement.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const removeArrayElement = (array, index) => [
```
Comment (vlabs) [inline src/utils/removeArrayElement.js]: :+1:

File: src/components/RuleSet.js (hunk 1 above)
Comment (jasrial) [reply src/components/RuleSet.js]: c1094254c81183e67d59b06792ccb374c021f5a0

File: src/components/RuleSet.js (hunk 2 above)
Comment (jasrial) [reply src/components/RuleSet.js]: 4836fb93ccaa7bc604fae1f687d774774290fe5c

**Pull request merged**